### PR TITLE
feat: Adding disable property to ColorPicker

### PR DESCRIPTION
**Pull request by palbizu**
## Description
Disabling ColorPicker

image

### Testing
Locally tested

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/components/src/color-picker/color-picker.component.ts [hunk 1]
```diff
@@ -94,6 +106,10 @@ export class ColorPickerComponent implements ControlValueAccessor, OnChanges {
     this.propagateControlValueChangeOnTouch = onTouch;
   }
 
+  public setDisabledState(isDisabled?: boolean): void {
```
Comment (jake-bassett) [inline projects/components/src/color-picker/color-picker.component.ts:109]: This is fine, but could also do `public setDisabledState(isDisabled: boolean = false): void {`

And remove the `??` check below.

File: projects/components/src/color-picker/color-picker.component.ts [hunk 2]
```diff
@@ -41,6 +46,9 @@ export class ColorPickerComponent implements ControlValueAccessor, OnChanges {
   @Input()
   public selected?: string;
 
+  @Input()
+  public disabled: boolean = false;
```
Comment (jake-bassett) [inline projects/components/src/color-picker/color-picker.component.ts:50]: There is alway a risk of the user passing in a `null` or `undefined` here, so it's safest to type this as optional `?`.

**Review by jake-bassett — DISMISSED** [review]

File: projects/components/src/color-picker/color-picker.component.ts (hunk 1 above)
Comment (aaron-steinfeld) [reply projects/components/src/color-picker/color-picker.component.ts:109]: careful - if these are called by angular and it sends in nulls, the current code would assign false and the suggestion would assign null. May not be relevant, but something to keep in mind when working in code that interfaces with angular.

Comment (palbizu) [reply projects/components/src/color-picker/color-picker.component.ts:109]: I prefer to keep this change to keep the consistency with the rest of CVA components.

File: projects/components/src/color-picker/color-picker.component.ts (hunk 2 above)
Comment (palbizu) [reply projects/components/src/color-picker/color-picker.component.ts]: done

File: projects/components/src/color-picker/color-picker.component.ts (hunk 1 above)
Comment (jake-bassett) [reply projects/components/src/color-picker/color-picker.component.ts:109]: Is there a concern for angular or other code we don't own calling this method? If so, my suggestion above is safer.

Comment (aaron-steinfeld) [reply projects/components/src/color-picker/color-picker.component.ts:109]: this method is from the CVA interface, angular forms calls it. I'm not sure why we default it at all thoguh, that's not part of angular's contract:
https://angular.io/api/forms/ControlValueAccessor#setdisabledstate

Comment (palbizu) [reply projects/components/src/color-picker/color-picker.component.ts]: I took a look at the documentation and it's no concern. I just tried to keep the consistency with the rest of the components but already changed.

**Review by jake-bassett — APPROVED** [review]

**Pull request merged**